vault/resource_gcp_secret_impersonated_account.go

```diff
@@ -0,0 +1,235 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+var (
+	gcpSecretImpersonatedAccountBackendFromPathRegex = regexp.MustCompile("^(.+)/impersonated-account/.+$")
+	gcpSecretImpersonatedAccountNameFromPathRegex    = regexp.MustCompile("^.+/impersonated-account/(.+)$")
+)
+
+func gcpSecretImpersonatedAccountResource() *schema.Resource {
+	return &schema.Resource{
+		Create: gcpSecretImpersonatedAccountCreate,
+		Read:   gcpSecretImpersonatedAccountRead,
+		Update: gcpSecretImpersonatedAccountUpdate,
+		Delete: gcpSecretImpersonatedAccountDelete,
+		Exists: gcpSecretImpersonatedAccountExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
Thanks for the question @robmonte! Yes, the comments above regarding the deprecation of `Create`, `Read` etc. are accurate, and we prefer using the `Context` functions over the deprecated operations. 

As for the wrapper methods, here's the documentation for these functions:

- `ReadContextWrapper` provides common read operations to the wrapped `schema.ReadContextFunc`. This wrapper method is used on all `ReadContext` methods to set some common schema fields to the TF state (for eg., each resource has a `namespace` field that needs to be read from Vault and set to the TF state). [Here is an example](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kubernetes_secret_backend.go#L26) of it being used in the codebase.

- `MountCreateContextWrapper` performs a minimum version requirement check prior to the wrapped `schema.CreateContextFunc.` This wrapper method should be wrapped on the `CreateContext` method if there is a minimum version check required for this feature. Since this is a `Vault v1.13.0` feature, I'm guessing we should add a minimum version check with Vault 1.13. This would look [something like this](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kubernetes_secret_backend.go#L25), and would require us to add a `provider.VaultVersion113` to the codebase as [done here](https://github.com/hashicorp/terraform-provider-vault/blob/main/internal/provider/meta.go#L36)

Reply:
`MustAddMountMigrationSchema` shouldn't be needed in this case, since it deals with adding specific fields to the schema in case the resource supports mount migration (only resources that actually mount a secret backend need these schema fields, so we should be good in this case)